services/crates/crates-size.service.js

```diff
@@ -0,0 +1,56 @@
+import prettyBytes from 'pretty-bytes'
+import { pathParams } from '../index.js'
+import { BaseCratesService, description } from './crates-base.js'
+
+export default class CratesSize extends BaseCratesService {
+  static category = 'size'
+  static route = {
+    base: 'crates/size',
+    pattern: ':crate/:version?',
+  }
+
+  static openApi = {
+    '/crates/size/{crate}': {
+      get: {
+        summary: 'Crates.io Size',
+        description,
+        parameters: pathParams({
+          name: 'crate',
+          example: 'rustc-serialize',
+        }),
+      },
+    },
+    '/crates/size/{crate}/{version}': {
+      get: {
+        summary: 'Crates.io Size (version)',
+        description,
+        parameters: pathParams(
+          {
+            name: 'crate',
+            example: 'rustc-serialize',
+          },
+          {
+            name: 'version',
+            example: '0.3.24',
+          },
+        ),
+      },
+    },
+  }
+
+  render({ size }) {
+    return {
+      label: 'size',
+      message: size ? prettyBytes(size) : 'unknown',
+      color: size ? 'blue' : 'lightgray',
+    }
```

Comment:
In general, this looks good.

This is a bit of a minor point but instead of this, could we handle the case where `size` is not set by validating in `handle()` and throwing an exception, and then `render()` is just renders the valid response.

So we check something like:

```js
if (this.constructor.getVersionObj(json).crate_size == null) {
    throw new InvalidResponse({ prettyMessage: 'unknown' })
}
```

in `handle` and then `render` just does

```js
render({ size }) {
  return {
    label: 'size',
    message: prettyBytes(size),
    color: 'blue'
  }
}
```

Reply:
Yep I agree and have made the changes, thanks for pointing that out